lib/ReadsUtils/ReadsUtilsImpl.py

```diff
@@ -642,6 +680,330 @@ def get_fq_stats(self, reads_object, file_path):
+        except urllib2.URLError as e:
+            raise ValueError("Failed to reach a server\nReason: %s" % e.reason)
+        else:
+            with closing(online_file):
+                with open(copy_file_path, 'wb') as output:
+                    shutil.copyfileobj(online_file, output)
+            self.log('Downloaded file to %s' % copy_file_path)
+
+    def _download_dropbox_link(self, file_url, copy_file_path):
+        """
+        _download_dropbox_link: dropbox download link handler
+                                file needs to be shared publicly 
+
+        params:
+        file_url: dropbox download link
+        copy_file_path: output file saving path
+
+        """
+        # translate dropbox URL for direct download
+        if "?" not in file_url:
+            force_download_link = file_url + '?raw=1'
+        else:
+            force_download_link = file_url.partition('?')[0] + '?raw=1'
+
+        self.log('Generating DropBox direct download link\n from: %s\n to: %s' % (file_url, force_download_link))
+        self._download_direct_download_link(force_download_link, copy_file_path)
+
+    def _download_ftp_link(self, file_url, copy_file_path):
+        """
+        _download_ftp_link: FTP download link handler
+                            URL fomat: ftp://anonymous:email@ftp_link 
+                                    or ftp://ftp_link
+                            defualt user_name: 'anonymous'
+                                    password: 'anonymous@domain.com'
+
+                            Note: Currenlty we only support anonymous FTP due to securty reasons.
+
+        params:
+        file_url: FTP download link
+        copy_file_path: output file saving path
+
+        """
+        print "=== WARNING ===\n=== WARNING ===\n"
+        print "FTP IS NOT A SECURIED PROTOCOL\nCREDENTIALS AND FILE CONTENTS WILL BE EXPOSED TO THE PUBLIC"
+        print "\n=== STOP THE PROCESS NOW IF YOU DO NOT WISH TO EXPOSE PRIVATE DATA ==="
+        time.sleep(60)
+
+        self.log('Connecting FTP link: %s' % file_url)
+        ftp_url_format = re.match(r'ftp://.*:.*@.*/.*', file_url)
+        # process ftp credentials
+        if ftp_url_format:
+            self.ftp_user_name = re.search('ftp://(.+?):', file_url).group(1)
+            if self.ftp_user_name.lower() != 'anonymous':
+                raise ValueError("Currently we only support anonymous FTP")
+            self.ftp_password = file_url.rpartition('@')[0].rpartition(':')[-1]
+            self.ftp_domain = re.search('ftp://.*:.*@(.+?)/', file_url).group(1)
+            self.ftp_file_path = file_url.partition('ftp://')[-1].partition('/')[-1].rpartition('/')[0]
+            self.ftp_file_name = re.search('ftp://.*:.*@.*/(.+$)', file_url).group(1)
+        else:
+            self.log('Setting anonymous FTP user_name and password')
+            self.ftp_user_name = 'anonymous'
+            self.ftp_password = 'anonymous@domain.com'
```

Comment:
The domain in password can be anything. The password can be any email format. The actual domain is parsed in the next step.